usaspending_api/awards/v2/filters/matview_filters.py

```diff
@@ -255,10 +255,23 @@ def keyword_parse(keyword):
                 queryset = queryset.filter(cfda_number__in=in_query)
 
         elif key == "naics_codes":
-            if value:
-                regex = f"^({'|'.join([str(elem) for elem in value])}).*"
+            if isinstance(value, list):
```

Comment:
Should the validation for the `naics_code` be in the filter code?

Reply:
The exception can be removed since TinyShield is already covering this. I was just trying to fail loudly in  the event both of these checks fail, mostly to not have the code look strange here.

Reply:
Personally would like to see the error removed. If we have the same error thrown in multiple places it turns into an easter egg hunt to find where it came from.

Reply:
Did this code get changed per the comment @BrianZito?
@sethstoudenmier if you had code that leveraged the 501 response, can you cite it here. Not sure if you can use it Brian, but would be good to distinguish "something unknown when wrong" (`500`), "an error condition we know about is being reported (`501`)"

Reply:
We  discussed this when we went over the code  in person. Did  we come to any  conclusions, Seth?

Reply:
There is a `NotImplementedException` which throws a 501, but in this case, I think an InvalidParamaterException is a correct response. My comment was around removing the thrown error all-together if we are confident that it is going to be caught by something such as TinyShield. Otherwise, I am fine with it here.